### PR TITLE
chore: expose jsonrpc types for subscription

### PR DESCRIPTION
**Pull request by yangby-cryptape**
At present, write a subscription client have to copy those codes or depend on the server crate.
It doesn't make sense.

### CODE REVIEW

**Review by zhangsoledad — APPROVED** [review]

**Review by quake — APPROVED** [review]

File: util/jsonrpc-types/src/subscription.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use serde::{Deserialize, Serialize};
+
+/// Specifies the topic which to be added as active subscription.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Hash)]
```
Comment (keroro520) [inline util/jsonrpc-types/src/subscription.rs:4]:
```suggestion
#[derive(Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
```

**Review by keroro520 — CHANGES_REQUESTED** [review]

File: util/jsonrpc-types/src/subscription.rs (hunk 1 above)
Comment (yangby-cryptape) [reply util/jsonrpc-types/src/subscription.rs]: DONE!

**Review by keroro520 — APPROVED** [review]

**Review by zhangsoledad — DISMISSED** [review]

**Review by doitian — APPROVED** [review]

**Comment by doitian** [comment]
0001-docs-rpc-fix-rpc-doc-for-Topic.patch

```diff
From e53a919638cc987037f750768564c4e59d522d62 Mon Sep 17 00:00:00 2001
From: ian <ian@nervos.org>
Date: Thu, 12 Nov 2020 10:16:18 +0800
Subject: [PATCH] docs(rpc): fix rpc doc for Topic

---
 devtools/doc/rpc.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/devtools/doc/rpc.py b/devtools/doc/rpc.py
index 06613bd1f..4578cce11 100755
--- a/devtools/doc/rpc.py
+++ b/devtools/doc/rpc.py
@@ -84,7 +84,10 @@ def write_method_signature(file, method_name, vars):
             file.write('    * `{}`: {}\n'.format(var.name, var.ty))
     else:
         file.write('* `{}()`\n'.format(method_name))
-    file.write('* result: {}\n'.format(vars[-1].ty))
+    if method_name == 'subscribe':
+        file.write('* result: `string`\n')
+    else:
+        file.write('* result: {}\n'.format(vars[-1].ty))
 
 
 class MarkdownParser():
@@ -674,7 +677,7 @@ class RPCDoc(object):
 
         if 'ckb_types/packed' in path:
             return
-        if path.split('/')[-1] in ['type.Result.html', 'struct.Subscriber.html', 'enum.SubscriptionId.html']:
+        if path.split('/')[-1] in ['type.Result.html', 'struct.Subscriber.html', 'enum.SubscriptionId.html', 'enum.Topic.html']:
             return
 
         with open(path) as file:
-- 
2.29.2
```

**Review by keroro520 — APPROVED** [review]

**Review by quake — APPROVED** [review]

**Review by zhangsoledad — APPROVED** [review]

**Comment by zhangsoledad** [comment]
bors r=quake,zhangsoledad

**Comment by doitian** [comment]
bors retry

**Review by doitian — APPROVED** [review]

**Comment by zhangsoledad** [comment]
bors retry

**Pull request merged**